### PR TITLE
Enabled usage of the $ntpsigndsocket parameter, for socket signing.

### PR DESCRIPTION
**Pull request by powertoaster**
I need to be able to sign NTP packets and this enables the parameter.

### CODE REVIEW

**Comment by bmjen** [comment]
@powertoaster This looks good but can you please remove the updates to the CHANGELOG and the metadata version bump? We will take care of that part when we prep the release. Thanks!

**Comment by powertoaster** [comment]
Done, thank you all for your help and patience with me.

File: README.markdown [hunk 1]
```diff
@@ -225,6 +225,10 @@ Tells Puppet to use non-standard minimal poll interval of upstream servers. Vali
 
 Tells Puppet to use non-standard maximal poll interval of upstream servers. Valid options: 3 to 16. Default option: undef, except FreeBSD (on FreeBSD `maxpoll` set 9 by default).
 
+####`ntpsigndsocket`
+
+Tells NTP to sign packets using the socket in the ntpsigndsocket path. NTP must be configured to sign sockets for this to work.
```
Comment (bmjen) [inline README.markdown]: Can you add a line here describing what the parameter should take and what the Default is? Similar to the descriptions provided by the other parameters around it.

**Comment by bmjen** [comment]
@powertoaster No problem. Thank you for contributing! I'm sorry I didn't catch this before my last comment. But I've made one more request in the README. Thanks again!

**Comment by powertoaster** [comment]
Added the requested comments, I want to thank everyone at Puppet for letting me flail around here a little. I have learned a great deal about GIT and contributing to a project in the process. Way more than I would have had one of you cleaned up my little mistakes for me.

Oh, and puppet rules.

**Comment by bmjen** [comment]
@powertoaster The README still needs a blurb about the Parameter input and default. Sorry for the back and forth. :(

**Comment by powertoaster** [comment]
I pushed it to my repo, I can see the changes there but for some reason they are not appearing here. I frankly am not sure how to force them, all of my other changes to the pull request code came over automatically. All I could do at this point is close the pull and redo it.

**Comment by bmjen** [comment]
@powertoaster Actually.. they appear to have updated, but apparently didn't resolve my comment. weird.

File: README.markdown [hunk 2]
```diff
@@ -225,6 +225,11 @@ Tells Puppet to use non-standard minimal poll interval of upstream servers. Vali
...
+Valid options a path to the socket directory, in the case of Samba it would be: ntpsigndsocket = usr/local/samba/var/lib/ntp_signd, there is no default.
```
Comment (bmjen) [inline README.markdown]: Typically.. you'd say that `Default value: undef` as is the case here.

**Comment by bmjen** [comment]
@powertoaster Apologies for nitpicking the docs, I just want to maintain consistency through the README when new parameters or features are added. Thanks for your patience!

**Comment by powertoaster** [comment]
Should be good to go

**Comment by bmjen** [comment]
Thanks @powertoaster !

**Pull request merged**